.github/workflows/pr.yml

```diff
@@ -1,38 +1,88 @@
 name: Checks
 on: [push, pull_request]
+env:
+  NODE_VERSION: 20
 jobs:
+  # Multiple jobs depend on having EVM and SVM artifacts available, so we upload them first in separate jobs.
+  upload-svm-artifacts:
+    name: Upload SVM artifacts
+    runs-on: ubuntu-latest
+    env:
+      ARTIFACTS_PATHS: |
+        target/idl
+        target/types
+        src/svm/assets
+        src/svm/clients
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
```

Comment:
Now that we’ve got more jobs, it might be worth caching `actions/checkout@v3` too, using something like `hashFiles('**/.git/**')`. We’re doing a v3 checkout in 7 different jobs, plus a v2 checkout in the publish step.